graylog2-server/src/main/java/org/graylog2/indexer/datastream/policy/actions/RolloverActionFormatter.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+package org.graylog2.indexer.datastream.policy.actions;
+
+import static org.graylog2.shared.utilities.StringUtils.f;
+
+public class RolloverActionFormatter {
+
+    private RolloverActionFormatter() {
+    }
+
+    /**
+     * Formats retention value in days using the proper OpenSearch time value format (e.g. 14d).
+     * <a href="https://opensearch.org/docs/latest/im-plugin/ism/policies/#transitions">...</a>
+     */
+    public static String formatDaysDuration(long days) {
```

Comment:
I think this could maybe also be more generic to take the duration as a parameter and set the unit accordingly?

Reply:
Thanks for the suggestion. After looking at this again, it looked like a nice candidate for an enum with simple formatter for each choice, so I broke those out into separate respective enums. Please let me know if you have any suggestions for improving this further. Added in https://github.com/Graylog2/graylog2-server/pull/18473/commits/5631edc29fbb6cec968e689bdcd98fbeb8497450.